src/subtyping.md

```diff
@@ -1,146 +1,127 @@
 # Subtyping and Variance
 
-Subtyping is a relationship between types that allows statically typed
-languages to be a bit more flexible and permissive.
+Rust uses lifetimes to track the relationships between borrows and ownership.
+However, a naive implementation of lifetimes would be either too restrictive,
+or permit undefined behaviour.
 
-Subtyping in Rust is a bit different from subtyping in other languages. This
-makes it harder to give simple examples, which is a problem since subtyping,
-and especially variance, is already hard to understand properly. As in,
-even compiler writers mess it up all the time.
+In order to allow flexible usage of lifetimes
+while also preventing their misuse, Rust uses a combination of **Subtyping** and **Variance**.
 
-To keep things simple, this section will consider a small extension to the
-Rust language that adds a new and simpler subtyping relationship. After
-establishing concepts and issues under this simpler system,
-we will then relate it back to how subtyping actually occurs in Rust.
+## Subtyping
 
-So here's our simple extension, *Objective Rust*, featuring three new types:
+Subtyping is the idea that one type can be a *subtype* of another.
 
-```rust
-trait Animal {
-    fn snuggle(&self);
-    fn eat(&mut self);
-}
+Let's define that `Sub` is a subtype of `Super` (we'll be using the notation `Sub: Super` throughout this chapter)
 
-trait Cat: Animal {
-    fn meow(&self);
-}
+What this is suggesting to us is that the set of *requirements* that `Super` defines
+are completely satisfied by `Sub`. `Sub` may then have more requirements.
 
-trait Dog: Animal {
-    fn bark(&self);
+An example of simple subtyping that exists in the language are [supertraits](https://doc.rust-lang.org/stable/book/ch19-03-advanced-traits.html?highlight=supertraits#using-supertraits-to-require-one-traits-functionality-within-another-trait)
+
+```rust
+use std::fmt;
+
+trait OutlinePrint: fmt::Display {
+    fn outline_print(&self) {
+        todo!()
+    }
 }
 ```
 
-But unlike normal traits, we can use them as concrete and sized types, just like structs.
+Here, we have that `OutlinePrint: fmt::Display` (`OutlinePrint` is a *subtype* of `Display`),
+because it has all the requirements of `fmt::Display`, plus the `outline_print` function.
 
-Now, say we have a very simple function that takes an Animal, like this:
+However, subtyping in traits is not that interesting in the case of Rust.
+Here in the nomicon, we're going to focus more with how subtyping interacts with **lifetimes**
+
+Take this example
```

Comment:
Personally I would lead the chapter with an example, *before even mentioning variance and subtyping*, showing an example that should compile but doesn't, where the user can see that it makes sense, and then drop *we can fix this with variance*